File: py-util/dayx.py
# from parse import *
from functools import reduce
from collections import defaultdict
import sys
import re
import json
import os
import types
import math

sys.path.append(os.getcwd() + "\py-util")
import util


def switchToTest():
    global filename
    print ("Testing ... ")
    filename = "..\\data\\test.txt"


def init(lines):
    retval = []
    for line in lines:
        pass
    retval = lines if len(retval) == 0 else retval
    return retval


def part1():
    answer = 0

    print("answer part 1:", answer)
    # assert answer in [0, 0], "answer is wrong " + str(answer)
    pass


def part2():
    answer = 0

    print("answer part 2:", answer)
    # assert answer in [0, 0], "answer is wrong " + str(answer)
    pass


filename = "..\\data\\dx_input.txt"
switchToTest()

abs_file_path = os.path.join(os.path.dirname(__file__), filename)
lines = open(abs_file_path, "r").readlines()
lines = list(map(lambda x: x.strip(), lines))

# print(*lines, sep="\n")

input = init(lines)

part1()
part2()

File: js-util/dayx.js
import * as tools from '../../js-util/tools.js';

const YEAR = the_year;
const baseDir = `${process.cwd()}\\advent${YEAR}`;
const dataDir = `${baseDir}\\data`;

function part1(lines) {
    let answer = 0;
    console.log(`Part 1: ${answer}`)
    const finalAnswer = undefined
    console.assert(finalAnswer === answer, `${answer} should have been ${finalAnswer}`);
}

function part2(lines) {
    let answer = 0;
    console.log(`Part 2: ${answer}`)
    const finalAnswer = undefined
    console.assert(finalAnswer === answer, `${answer} should have been ${finalAnswer}`);
}

function main() {
    let inputFile = `${dataDir}\\dx_input.txt`;
    inputFile = `${baseDir}\\data\\test.txt`;

    console.log("Input File: " + inputFile);
    const lines = tools.readFileFromLocal(inputFile).split(/\r?\n/).filter(x => x);
    if (lines[lines.length - 1].length == 0) lines.pop();

    part1(lines);
    part2(lines);
}

main();